adaptive/src/sharedTest/kotlin/com/google/accompanist/adaptive/FoldAwareColumnTest.kt

```diff
@@ -0,0 +1,232 @@
+    private lateinit var foldBoundsPx: RectF
+    private var firstSpacerDp = 0.dp
+    private var secondSpacerDp = 0.dp
+    private var secondSpacerTopPx = 0f
+    private var secondSpacerBoundsPx = RectF()
+
+    @Test
+    fun hinge_places_correctly() {
+        setUp()
+
+        assertEquals(foldBoundsPx.bottom, secondSpacerTopPx)
+    }
+
+    @Test
+    fun separating_fold_places_correctly() {
+        setUp(foldSizePx = 0)
+
+        assertEquals(foldBoundsPx.bottom, secondSpacerTopPx)
+    }
+
+    @Test
+    fun no_fold_places_normally() {
+        setUp(includeFold = false)
+
+        composeTestRule.onNodeWithTag(testTag).assertTopPositionInRootIsEqualTo(firstSpacerDp)
+    }
+
+    @Test
+    fun non_separating_fold_places_normally() {
+        setUp(foldSizePx = 0, foldState = FoldingFeature.State.FLAT)
+
+        composeTestRule.onNodeWithTag(testTag).assertTopPositionInRootIsEqualTo(firstSpacerDp)
+    }
+
+    @Test
+    fun vertical_fold_places_normally() {
+        setUp(foldOrientation = FoldingFeature.Orientation.VERTICAL)
+
+        composeTestRule.onNodeWithTag(testTag).assertTopPositionInRootIsEqualTo(firstSpacerDp)
+    }
+
+    @Test
+    fun ignore_fold_modifier_places_correctly() {
+        setUp(secondSpacerModifier = Modifier.ignoreFold())
+
+        composeTestRule.onNodeWithTag(testTag).assertTopPositionInRootIsEqualTo(firstSpacerDp)
+    }
+
+    @Test
+    fun even_fold_padding_modifier_places_correctly_around_fold() {
+        val foldPadding = 20.dp
+        setUp(foldPadding = PaddingValues(vertical = foldPadding))
+
+        with(density) {
+            assertEquals(foldBoundsPx.bottom + foldPadding.toPx(), secondSpacerTopPx)
+        }
+    }
+
+    @Test
+    fun uneven_fold_padding_modifier_places_correctly_around_fold() {
+        val foldPaddingBottom = 40.dp
+        setUp(foldPadding = PaddingValues(top = 15.dp, bottom = 40.dp))
+
+        with(density) {
+            assertEquals(foldBoundsPx.bottom + foldPaddingBottom.toPx(), secondSpacerTopPx)
+        }
+    }
+
+    @Test
+    fun layout_bounds_align_with_child_bounds_normally() {
+        setUp(includeFold = false)
+
+        val layoutBottomPx = composeTestRule.onRoot().fetchSemanticsNode().boundsInWindow.bottom
+
+        assertEquals(layoutBottomPx, secondSpacerBoundsPx.bottom)
+    }
+
+    @Test
+    fun layout_bounds_contain_child_bounds_with_separating_fold() {
+        setUp(firstSpacerPct = 0.1f, secondSpacerPct = 0.1f)
+
+        val layoutBottomPx = composeTestRule.onRoot().fetchSemanticsNode().boundsInWindow.bottom
+
+        assert(secondSpacerBoundsPx.bottom <= layoutBottomPx)
+    }
+
+    @Test
+    fun layout_bounds_align_with_child_bounds_when_pushed_below_separating_fold() {
+        setUp()
+
+        val layoutBottomPx = composeTestRule.onRoot().fetchSemanticsNode().boundsInWindow.bottom
+
+        assertEquals(layoutBottomPx, secondSpacerBoundsPx.bottom)
+    }
+
+    private fun setUp(
```

Comment:
Great point! I ended up splitting up the `setUp` function because I think it was trying to do too much in one spot - instead of making it a `@Before` method, we now just call set test content and simulate a fold as necessary.

Does the new format look good?

Reply:
Yep!